tests/UnifiedSpecTests/Constraint/MatchesTest.php

```diff
@@ -69,6 +71,10 @@ public function testOperatorExists(): void
 
     public function testOperatorType(): void
     {
+        if (version_compare(phpversion(), '8.4', '>=')) {
+            $this->markTestIncomplete('Test fails on PHP 8.4 due to deprecations');
+        }
+
```

Comment:
Nit: I prefer using the `PHP_VERSION_ID` for performance reason.

```suggestion
        if (PHP_VERSION_ID >= 8_04_00) {
            $this->markTestIncomplete('Test fails on PHP 8.4 due to deprecations');
        }

```

But the best is to use the [`#[RequiresPhp('>= 8.4')]` attribute](https://docs.phpunit.de/en/10.5/attributes.html#requiresphp).

Reply:
Ah, good point. There's also `#[RequiresPhpExtension]` that we can use in another tests that depends on the extension version. I wonder if we can also leverage this mechanism for server version checks - I'll look into it.

Reply:
Might be tricky for server version checks as it'd need to use a persistent client connection. But there may be an opportunity there to streamline everything with a service that captures all of that information when PHPUnit is first loaded (fetching the URI from environment variables).